api/pkg/apis/v1alpha1/managers/catalogs/catalogs-manager.go

```diff
@@ -152,6 +153,73 @@ func (m *CatalogsManager) UpsertState(ctx context.Context, name string, state mo
 		return err
 	}
 
+	metadata := map[string]interface{}{
+		"namespace": state.ObjectMeta.Namespace,
+		"group":     model.FederationGroup,
+		"version":   "v1",
+		"resource":  "catalogs",
+		"kind":      "Catalog",
+	}
+
+	if state.Spec != nil {
+		rootResource := state.Spec.RootResource
+		if rootResource != "" {
+			log.Debugf(" M (Catalogs): catalog root resource: %s, catalog: %s", rootResource, name)
+			resourceName := "catalogcontainers"
+			kind := "CatalogContainer"
+			containerMetadata := map[string]interface{}{
+				"version":   "v1",
+				"group":     model.FederationGroup,
+				"resource":  resourceName,
+				"namespace": state.ObjectMeta.Namespace,
+				"kind":      kind,
+			}
+			getRequest := states.GetRequest{
+				ID:       rootResource,
+				Metadata: containerMetadata,
+			}
+			_, err = m.StateProvider.Get(ctx, getRequest)
+			if err != nil {
+				log.Debugf(" M (Catalogs): get catalog container %s, err %v", rootResource, err)
+			}
+			cErr, ok := err.(v1alpha2.COAError)
+			if ok && cErr.State == v1alpha2.NotFound {
+				objectMeta := model.ObjectMeta{Namespace: state.ObjectMeta.Namespace, Name: rootResource}
+				label_key := os.Getenv("LABEL_KEY")
```

Comment:
What's this? Why is a system-wide setting? What kind of scenarios do we try to enable?

Reply:
Sorry, haishi. This is a draft PR based on Lency's https://github.com/eclipse-symphony/symphony/pull/297.
This PR's purpose is to offer users an option to add label to the materialized resources so that they can filter them by the label. You can review lency's first.

Reply:
please rebase latest main.